prow/cmd/generic-autobumper/main.go

```diff
@@ -19,47 +19,467 @@ package main
 import (
 	"fmt"
 	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
 
 	flag "github.com/spf13/pflag"
 
 	"github.com/sirupsen/logrus"
 
+	imagebumper "k8s.io/test-infra/experiment/image-bumper/bumper"
 	"k8s.io/test-infra/prow/cmd/generic-autobumper/bumper"
 
 	"sigs.k8s.io/yaml"
 )
 
-func parseOptions() (*bumper.Options, error) {
+const (
+	latestVersion          = "latest"
+	upstreamVersion        = "upstream"
+	upstreamStagingVersion = "upstream-staging"
+	tagVersion             = "vYYYYMMDD-deadbeef"
+	defaultUpstreamURLBase = "https://raw.githubusercontent.com/kubernetes/test-infra/master"
+)
+
+var (
+	tagRegexp    = regexp.MustCompile("v[0-9]{8}-[a-f0-9]{6,9}")
+	imageMatcher = regexp.MustCompile(`(?s)^.+image:(.+):(v[a-zA-Z0-9_.-]+)`)
+)
+
+var _ bumper.PRHandler = (*client)(nil)
+
+type client struct {
+	o        *Options
+	images   map[string]string
+	versions map[string][]string
+}
+
+// Changes returns a slice of functions, each one does some stuff, and
+// returns commit message for the changes
+func (c *client) Changes() []func() (string, error) {
+	return []func() (string, error){
+		func() (string, error) {
+			var err error
+			c.images, err = UpdateReferences(c.o)
+			if err != nil {
+				return "", fmt.Errorf("failed to update image references: %w", err)
+			}
+
+			c.versions, err = getVersionsAndCheckConsistency(c.o.Prefixes, c.images)
+			if err != nil {
+				return "", err
+			}
+
+			var body string
+			for _, prefix := range c.o.Prefixes {
+				body = body + generateSummary(prefix.Name, prefix.Repo, prefix.Prefix, prefix.Summarise, c.images) + "\n\n"
+			}
+
+			return body, nil
+		},
+	}
+}
+
+// PRTitleBody returns the body of the PR, this function runs after each commit
+func (c *client) PRTitleBody() (string, string, error) {
+	return makeCommitSummary(c.o.Prefixes, c.versions), "", nil
+}
+
+// Options is the options for autobumper operations.
+type Options struct {
```

Comment:
suspect this doesn't need to be exported either

Reply:
Made it private